java-client/src/main/java/org/opensearch/client/opensearch/snapshot/GetSnapshotResponse.java

```diff
@@ -50,208 +57,130 @@
 // typedef: snapshot.get.Response
 
 @JsonpDeserializable
+@Generated("org.opensearch.client.codegen.CodeGenerator")
 public class GetSnapshotResponse implements PlainJsonSerializable {
-    private final List<SnapshotResponseItem> responses;
 
+    @Nonnull
     private final List<SnapshotInfo> snapshots;
 
-    private final int total;
-
-    private final int remaining;
-
     // ---------------------------------------------------------------------------------------------
 
     private GetSnapshotResponse(Builder builder) {
-
-        this.responses = ApiTypeHelper.unmodifiable(builder.responses);
-        this.snapshots = ApiTypeHelper.unmodifiable(builder.snapshots);
-        this.total = ApiTypeHelper.requireNonNull(builder.total, this, "total");
-        this.remaining = ApiTypeHelper.requireNonNull(builder.remaining, this, "remaining");
-
+        this.snapshots = ApiTypeHelper.unmodifiableRequired(builder.snapshots, this, "snapshots");
     }
 
-    public static GetSnapshotResponse of(Function<Builder, ObjectBuilder<GetSnapshotResponse>> fn) {
+    public static GetSnapshotResponse of(Function<GetSnapshotResponse.Builder, ObjectBuilder<GetSnapshotResponse>> fn) {
         return fn.apply(new Builder()).build();
     }
 
     /**
-     * API name: {@code responses}
+     * This method is left for compatibility. Always returns an empty list.
      */
+    @Deprecated
     public final List<SnapshotResponseItem> responses() {
-        return this.responses;
+        return new ArrayList<>();
     }
 
     /**
-     * API name: {@code snapshots}
+     * Required - API name: {@code snapshots}
      */
+    @Nonnull
     public final List<SnapshotInfo> snapshots() {
         return this.snapshots;
     }
 
     /**
-     * Required - The total number of snapshots that match the request when ignoring
-     * size limit or after query parameter.
+     * * This method is left for compatibility. Always returns 0.
      * <p>
      * API name: {@code total}
      */
+    @Deprecated
     public final int total() {
-        return this.total;
+        return 0;
```

Comment:
May be:

```suggestion
        return snapshots().size();
```

Reply:
Could also rather than removing the fields just make them optional, and just have the minor "breakage"/inconvenience of them going from `int` to `Integer`